server/src/test/java/org/elasticsearch/cluster/coordination/PublicationTransportHandlerTests.java

```diff
@@ -219,13 +226,13 @@ public RecyclerBytesStreamOutput newNetworkBytesStream() {
 
             final List<DiscoveryNode> allNodes = new ArrayList<>();
             while (allNodes.size() < 10) {
-                allNodes.add(
-                    new DiscoveryNode(
-                        "node-" + allNodes.size(),
-                        buildNewFakeTransportAddress(),
-                        VersionUtils.randomCompatibleVersion(random(), Version.CURRENT)
-                    )
+                var node = new DiscoveryNode(
+                    "node-" + allNodes.size(),
+                    buildNewFakeTransportAddress(),
+                    VersionUtils.randomCompatibleVersion(random(), Version.CURRENT)
                 );
+                allNodes.add(node);
+                nodeTransports.put(node, node.getVersion().transportVersion);
```

Comment:
This is not good, but how else do we get a transportversion from the corresponding version, if not through the field on `Version`?

Reply:
Can we just use a random transport version?

Reply:
It ideally needs to be the same or similar to the transportversion used for that same Version though

Reply:
I would expect that we can use any compatible transport version. We are only serialising things in this test, we don't even need to deserialise them AFAICT.